hpat/hiframes/pd_dataframe_ext.py

```diff
@@ -1259,6 +1259,39 @@ def lower_mean_dummy(context, builder, sig, args):
     return out_obj._getvalue()
 
 
+@overload_method(DataFrameType, 'median')
+def median_overload(df, axis=None, skipna=None, level=None, numeric_only=None):
+    # TODO: kwargs
+    # TODO: avoid dummy and generate func here when inlining is possible
+    def _impl(df, axis=None, skipna=None, level=None, numeric_only=None):
+        return hpat.hiframes.pd_dataframe_ext.median_dummy(df)
+
+    return _impl
+
+
+def median_dummy(df, n):
+    return df
+
+
+@infer_global(median_dummy)
+class MedianDummyTyper(AbstractTemplate):
+    def generic(self, args, kws):
+        df = args[0]
+        # TODO: ignore non-numerics
+        # output is float64 series with column names as string index
+        out = SeriesType(
+            types.float64, types.Array(types.float64, 1, 'C'),
+            string_array_type)
+        return signature(out, *args)
+
+
+@lower_builtin(median_dummy, types.VarArg(types.Any))
+def lower_median_dummy(context, builder, sig, args):
+    out_obj = cgutils.create_struct_proxy(
```

Comment:
Did you try to move this into line 1267?

Reply:
See comment in line 1265: with https://github.com/numba/numba/pull/4208 we can now simplify all this and get rid of the dummy construct. Maybe this should be done in an extra PR converting all the dummies in this file.

Reply:
@fschlimb Could you please point exact link to the material or just copy part of it here to help to find it?

Reply:
@shssf What material are you asking for? I think all I know is this PR...
As far as I understand one issue with using `@overload` is that (before the above PR to numba) the code in the `@overload` could not be inlined by numba. This in turn inhibits optimizations that numba/llvm can do with the current code (like loop-fusion). With `@overload(inline=always)` (or so) numba will now inline the code from an `@overload` and so loops over several series or several loops over the same series can potentially get fused. Such compiler opts are a major source of HPAT's performance advantages.